commit 2da796e74860394b745e12d55473509fafec2c04
Author: John Levon <john.levon@joyent.com>
Date:   2019-07-23T19:28:36+00:00 (3 months ago)
    
    OS-7790 enable smatch build for illumos-joyent (fix gcc7 build)
    Reviewed by: Tim Foster <tim.foster@joyent.com>
    Approved by: Tim Foster <tim.foster@joyent.com>

diff --git a/usr/contrib/freebsd/dev/nvme/nvme.h b/usr/contrib/freebsd/dev/nvme/nvme.h
index 73d4e2d2db..c7f6496426 100644
--- a/usr/contrib/freebsd/dev/nvme/nvme.h
+++ b/usr/contrib/freebsd/dev/nvme/nvme.h
@@ -26,6 +26,8 @@
  * SUCH DAMAGE.
  *
  * $FreeBSD$
+ *
+ * Copyright 2019 Joyent, Inc.
  */
 
 #ifndef __NVME_H__
@@ -1076,7 +1078,10 @@ struct nvme_health_information_page {
 	uint8_t			reserved2[296];
 } __packed __aligned(4);
 
+/* Currently sparse/smatch incorrectly packs this struct in some situations. */
+#ifndef __CHECKER__
 _Static_assert(sizeof(struct nvme_health_information_page) == 512, "bad size for nvme_health_information_page");
+#endif
 
 struct nvme_firmware_page {
 
